### PR TITLE
Native Attributes defined prior to custom

### PR DESCRIPTION
**Pull request by nakhbari**
[Review on Reviewable](https://reviewable.io/reviews/sonatype/codestyle/6)

### CODE REVIEW

File: Kaos-Frontend-Coding-Standards.md [hunk 1]
```diff
@@ -103,6 +103,12 @@ At Sonatype we value the stability and maintainability of the code base while st
   }
   FooController.$inject = ['$state'];
 ```
+#### HTML
```
Comment (whyjustin) [inline Kaos-Frontend-Coding-Standards.md]: Why subsubsubsub header?

File: Kaos-Frontend-Coding-Standards.md [hunk 2]
```diff
@@ -103,6 +103,12 @@ At Sonatype we value the stability and maintainability of the code base while st
...
+* Native attributes should be defined prior to custom attributes
```
Comment (whyjustin) [inline Kaos-Frontend-Coding-Standards.md]: Do we care if id comes first?

File: Kaos-Frontend-Coding-Standards.md [hunk 3]
```diff
@@ -103,6 +103,12 @@ At Sonatype we value the stability and maintainability of the code base while st
...
+  * This increases readabilities due to the ordering/organizing of the attributes
+```html
+  <textarea id="foo" class="bar" name="name" rows="2" ng-model="baz" ng-maxlength="255"></textarea>
```
Comment (whyjustin) [inline Kaos-Frontend-Coding-Standards.md]: Do we care if different attribute types are on different lines?

File: Kaos-Frontend-Coding-Standards.md (hunk 1 above)
Comment (nakhbari) [inline Kaos-Frontend-Coding-Standards.md]: Wanted to be under AngularJS Development and the sizing looks better.

File: Kaos-Frontend-Coding-Standards.md (hunk 2 above)
Comment (nakhbari) [inline Kaos-Frontend-Coding-Standards.md]: Was going to add that rule in a separate subsequent PR/change

File: Kaos-Frontend-Coding-Standards.md (hunk 3 above)
Comment (nakhbari) [inline Kaos-Frontend-Coding-Standards.md]: I don't have an opinion regarding that style.

**Comment by whyjustin** [comment]
+1

File: Kaos-Frontend-Coding-Standards.md (hunk 2 above)
Comment (mpiggott) [inline Kaos-Frontend-Coding-Standards.md]: What are native vs custom attributes

Comment (nakhbari) [inline Kaos-Frontend-Coding-Standards.md]: Native: ex. ```id``` ```class```
Custom: ex. Angular Attributes ```ng-model```

**Comment by seminolas** [comment]
+0. TIL 'readability' can be pluralised.

**Comment by mpiggott** [comment]
I'm not sure this is beneficial, logically to me I believe it makes more sense to group together by purpose.  For example `ng-class` and `class` are more appropriate beside each other than arbitrarily separated.

**Comment by whyjustin** [comment]
> I'm not sure this is beneficial, logically to me I believe it makes more sense to group together by purpose.

This makes sense, I'm retracting my +1 until we can think through some of the implications and the best approach.

**Comment by nakhbari** [comment]
@mpiggott I +1 your -1. I will close this and create separate proposal, proposing that we keep key attributes such as ```id``` and ```class``` (including their Angular counterparts) at the beginning of the definition.

**Comment by nakhbari** [comment]
New proposal https://github.com/sonatype/codestyle/pull/9

**Pull request closed without merging**